api/routes/graphs.py

```
@@ -225,3 +227,46 @@ async def delete_graph(request: Request, graph_id: str):
+    try:
+        logging.info("Received request to update user rules for graph: %s", graph_id)
+        logging.info("User rules content length: %d", len(data.user_rules))
+        full_graph_id = _graph_name(request.state.user_id, graph_id)
+        logging.info("Full graph_id: %s", full_graph_id)
+        await set_user_rules(full_graph_id, data.user_rules)
+        logging.info("User rules updated successfully for graph: %s", graph_id)
+        return JSONResponse(content={"success": True, "user_rules": data.user_rules})
+    except GraphNotFoundError:
+        logging.error("Graph not found: %s", graph_id)
+        return JSONResponse(content={"error": "Database not found"}, status_code=404)
+    except Exception as e:
```

Ruff (0.14.10)

Do not catch blind exception: `Exception`

(BLE001)